pkg/fleets/controller_test.go

```diff
@@ -244,50 +251,202 @@ func TestControllerSyncFleet(t *testing.T) {
 		agtesting.AssertEventContains(t, m.FakeRecorder.Events, "ScalingGameServerSet")
 		agtesting.AssertEventContains(t, m.FakeRecorder.Events, "CreatingGameServerSet")
 	})
+
+	t.Run("error on getting fleet", func(t *testing.T) {
+		c, _ := newFakeController()
+		c.fleetLister = &fakeFleetListerWithErr{}
+
+		err := c.syncFleet("default/fleet-1")
+		assert.EqualError(t, err, "error retrieving fleet fleet-1 from namespace default: err-from-namespace-lister")
+	})
+
+	t.Run("error on getting list of GS", func(t *testing.T) {
+		f := defaultFixture()
+		c, m := newFakeController()
+		c.gameServerSetLister = &fakeGSListerWithErr{}
+
+		m.AgonesClient.AddReactor("list", "fleets", func(action k8stesting.Action) (bool, runtime.Object, error) {
+			return true, &agonesv1.FleetList{Items: []agonesv1.Fleet{*f}}, nil
+		})
+
+		_, cancel := agtesting.StartInformers(m, c.fleetSynced, c.gameServerSetSynced)
+		defer cancel()
+
+		err := c.syncFleet("default/fleet-1")
+		assert.EqualError(t, err, "error listing gameserversets for fleet fleet-1: random-err")
+	})
+
+	t.Run("fleet not found", func(t *testing.T) {
+		c, _ := newFakeController()
+
+		err := c.syncFleet("default/fleet-1")
+		assert.Nil(t, err)
+	})
+
+	t.Run("fleet invalid strategy type", func(t *testing.T) {
+		f := defaultFixture()
+		c, m := newFakeController()
+		f.Spec.Strategy.Type = "invalid-strategy-type"
+
+		gsSet := f.GameServerSet()
+		// make gsSet.Spec.Template and f.Spec.Template different in order to make 'rest' list not empty
+		gsSet.Spec.Template.ClusterName = "qqqqqqqqqqqqqqqqqqq"
+
+		m.AgonesClient.AddReactor("list", "fleets", func(action k8stesting.Action) (bool, runtime.Object, error) {
+			return true, &agonesv1.FleetList{Items: []agonesv1.Fleet{*f}}, nil
+		})
+
+		m.AgonesClient.AddReactor("list", "gameserversets", func(action k8stesting.Action) (bool, runtime.Object, error) {
+			return true, &agonesv1.GameServerSetList{Items: []agonesv1.GameServerSet{*gsSet}}, nil
+		})
+
+		_, cancel := agtesting.StartInformers(m, c.fleetSynced, c.gameServerSetSynced)
+		defer cancel()
+
+		err := c.syncFleet("default/fleet-1")
+		assert.EqualError(t, err, "unexpected deployment strategy type: invalid-strategy-type")
+	})
+
+	t.Run("error on deleteEmptyGameServerSets", func(t *testing.T) {
+		f := defaultFixture()
+		c, m := newFakeController()
+
+		gsSet := f.GameServerSet()
+		// make gsSet.Spec.Template and f.Spec.Template different in order to make 'rest' list not empty
+		gsSet.Spec.Template.ClusterName = "qqqqqqqqqqqqqqqqqqq"
```

Comment:
```suggestion
		gsSet.Spec.Template.ClusterName = strings.Repeat("q", 19)
```

Reply:
I see it is not resolved

Reply:
Can we use a const for this and above?

Reply:
Yes, it's not resolved because I prefer to keep it as is. Honestly, I don't see any difference between a pure string and strings.Repeat(). In my implementation, it's easier to notice that this is an invalid value.
According to constants - I prefer to avoid them in tests and this is just a random "invalid" sequence of characters. This is not that type of stuff which is worth much attention, in my opinion.

Reply:
Yes, it might be not crucial here. Just it looks nicer.